### ISSUE TITLE
Spectator mode for multiplayer games (or skirmish with only AI players)

### ISSUE THREAD

**Issue by akien-mga** [OpenDungeons/OpenDungeons#666]
That would be a nice addition for a later milestone, especially for our shy players who don't want to compete against us because we know the core elements on the gameplay and how to exploit balancing issues :-p

Setting to "Later" for now, though I wouldn't be surprised to see @hwoarangmy grabbing this one for 0.6.0 together with the dedicated server stuff ;)

**Comment by hwoarangmy** [comment]
> Setting to "Later" for now, though I wouldn't be surprised to see @hwoarangmy grabbing this one for 0.6.0 together with the dedicated server stuff ;)

Depends. That would need more specifications. First, do we want to use specific seats (ie need to define observers in the level) ?
Second, what do an observer sees (everything? Only what one of the player sees) ?

**Comment by akien-mga** [comment]
I would make observers all-seeing, and not using seats defined in the level. IMO they should not be able to participate in the player chat (if they are all-seeing they could spoil the game), but maybe they could have a separate spectator chat to discuss between observers.

**Comment by Bertram25** [comment]
> I would make observers all-seeing, and not using seats defined in the level. IMO they should not be able to participate in the player chat (if they are all-seeing they could spoil the game), but maybe they could have a separate spectator chat to discuss between observers.

That point of view is relevant in the player's point but less technically, Eh eh.
This means the server will have to send every update data to every spectator and that may (or may not) kill performance. This needs investigating.

I'm still convinced it is something we should add, but maybe not until we have an option to disable them when hosting.
On the client side, the client would simply tell the server it connects as a specator and thus has no interaction whatsoever in game, except for chat. The game mode would also have to handle it by hiding the relevant widgets.

Well, definitely something for later, IMHO.

**Comment by hwoarangmy** [comment]
I agree it is for later. Moreover, that may also imply to change the server logic. ATM, events are only sent to relevant players. If observers can see everything, they should be able to see traps and everything. Well, might not be too difficult but as @Bertram25 pointed, it may hurt performances pretty much since observers would have vision on the whole map. It will be as long as opening the map in the editor...
